### ISSUE TITLE
caffe loss=0 always

### ISSUE THREAD

**Issue by maxadda** [weiliu89/caffe#9]
Hi, @weiliu89,I ran ssd in another dataset and then I found that the loss is always 0. Do you have any idea of such a problem?

Here are some results presented of the first 130 iterations:

> I0628 12:15:56.185905  3080 solver.cpp:231] Iteration 0, loss = 0
I0628 12:15:56.185936  3080 solver.cpp:247]     Train net output #0: mbox_loss = 0 (* 1 = 0 loss)
I0628 12:15:56.185964  3080 sgd_solver.cpp:106] Iteration 0, lr = 0.001
I0628 12:16:04.980218  3080 solver.cpp:231] Iteration 10, loss = 0
I0628 12:16:04.980249  3080 solver.cpp:247]     Train net output #0: mbox_loss = 0 (* 1 = 0 loss)
I0628 12:16:05.367525  3080 sgd_solver.cpp:106] Iteration 10, lr = 0.001
I0628 12:16:14.537703  3080 solver.cpp:231] Iteration 20, loss = 0
I0628 12:16:14.537734  3080 solver.cpp:247]     Train net output #0: mbox_loss = 0 (* 1 = 0 loss)
I0628 12:16:14.926493  3080 sgd_solver.cpp:106] Iteration 20, lr = 0.001
I0628 12:16:23.177091  3080 solver.cpp:231] Iteration 30, loss = 0
I0628 12:16:23.177120  3080 solver.cpp:247]     Train net output #0: mbox_loss = 0 (* 1 = 0 loss)
I0628 12:16:23.557282  3080 sgd_solver.cpp:106] Iteration 30, lr = 0.001
I0628 12:16:32.177230  3080 solver.cpp:231] Iteration 40, loss = 0
I0628 12:16:32.177299  3080 solver.cpp:247]     Train net output #0: mbox_loss = 0 (* 1 = 0 loss)
I0628 12:16:32.177320  3080 sgd_solver.cpp:106] Iteration 40, lr = 0.001
I0628 12:16:41.230350  3080 solver.cpp:231] Iteration 50, loss = 0
I0628 12:16:41.230391  3080 solver.cpp:247]     Train net output #0: mbox_loss = 0 (* 1 = 0 loss)
I0628 12:16:41.602286  3080 sgd_solver.cpp:106] Iteration 50, lr = 0.001
I0628 12:16:49.292162  3080 solver.cpp:231] Iteration 60, loss = 0
I0628 12:16:49.292201  3080 solver.cpp:247]     Train net output #0: mbox_loss = 0 (* 1 = 0 loss)
I0628 12:16:49.687913  3080 sgd_solver.cpp:106] Iteration 60, lr = 0.001
I0628 12:16:57.399756  3080 solver.cpp:231] Iteration 70, loss = 0
I0628 12:16:57.399794  3080 solver.cpp:247]     Train net output #0: mbox_loss = 0 (* 1 = 0 loss)
I0628 12:16:57.823508  3080 sgd_solver.cpp:106] Iteration 70, lr = 0.001
I0628 12:17:04.994513  3080 solver.cpp:231] Iteration 80, loss = 0
I0628 12:17:04.994612  3080 solver.cpp:247]     Train net output #0: mbox_loss = 0 (* 1 = 0 loss)
I0628 12:17:05.400712  3080 sgd_solver.cpp:106] Iteration 80, lr = 0.001
I0628 12:17:12.449678  3080 solver.cpp:231] Iteration 90, loss = 0
I0628 12:17:12.449715  3080 solver.cpp:247]     Train net output #0: mbox_loss = 0 (* 1 = 0 loss)
I0628 12:17:12.865375  3080 sgd_solver.cpp:106] Iteration 90, lr = 0.001
I0628 12:17:20.008211  3080 solver.cpp:231] Iteration 100, loss = 0
I0628 12:17:20.008249  3080 solver.cpp:247]     Train net output #0: mbox_loss = 0 (* 1 = 0 loss)
I0628 12:17:20.417125  3080 sgd_solver.cpp:106] Iteration 100, lr = 0.001
I0628 12:17:27.465800  3080 solver.cpp:231] Iteration 110, loss = 0
I0628 12:17:27.465837  3080 solver.cpp:247]     Train net output #0: mbox_loss = 0 (* 1 = 0 loss)
I0628 12:17:27.878662  3080 sgd_solver.cpp:106] Iteration 110, lr = 0.001
I0628 12:17:34.952018  3080 solver.cpp:231] Iteration 120, loss = 0
I0628 12:17:34.952057  3080 solver.cpp:247]     Train net output #0: mbox_loss = 0 (* 1 = 0 loss)
I0628 12:17:35.360599  3080 sgd_solver.cpp:106] Iteration 120, lr = 0.001
I0628 12:17:42.428278  3080 solver.cpp:231] Iteration 130, loss = 0

**Comment by weiliu89** [comment]
I am guessing you didn't prepare the dataset well. I would suggest you run this [ipynb](https://github.com/weiliu89/caffe/blob/ssd/examples/ssd.ipynb) to debug.

**Comment by maxadda** [comment]
Thank you! It's a problem of data format. I changed coordinates from float to int in .xml files and then it worked! @weiliu89

**Comment by JackieLeeTHU11** [comment]
@maxadda Hello, I have a same problem with you. What do you mean to change the coordinates from float to int.  change the format of "xmin, ymin, xmax, ymax" to be int? I change them to int, the problem of mbox_loss=0 does not disappear.

**Comment by catsdogone** [comment]
@JackieLeeTHU11 Have you solve this problem？I have the same problem. Could you give me some advise ? Thank  you.

**Comment by JackieLeeTHU11** [comment]
@catsdogone maybe the problem is the grount truth. I think the  "xmin, ymin, xmax, ymax" should be relative value (range from 0 to 1.0). I mean, xmin=xmin/image_width, ymin=ymin/image_hight. When I do this, the 'mbox_loss=0' disappeared. Please try it.
